Added a bunch of alt clicks / ctrl click behaviours.

KathyRyals opened the pull request:
## This PR is made in honor of SierraKomodo, who just learned he needed to actually press the submit button to submit a review.

## About The Pull Request
You can now Alt Click to eject people from Scanners / Cryotubes / sleepers. You can now CtrlClick to toggle on and off a cryotube.
Edit : Alt click to eject beakers from Chem dispenser and Chem master.
Edit Edit : Alt click to eject beakers from freezer / heater / grinder. Ctrl Click to grind. CtrlAlt Click to eject pills from the grinder.
## Why It's Good For The Game
QoL changes for people who don't like using the UI / right clicking the scanner, then drop down, the click eject scanner. 

## Changelog

:cl: Kathy Ryals
tweak: You can now Alt Click to eject people from scanners, cryotubes, or sleepers.
tweak: You can now Alt Click to eject a beaker from a chem master, dispenser, grinder, heater, or cooler.
tweak: You can now Ctrl Click a cryotube to turn it on and off.
tweak: You can now Ctrl Click a grinder to grind.
tweak: You can now CtrlAlt Click a grinder to eject the contents.
/:cl:

SierraKomodo reviewed (changes requested):
These need checks for distance, if the user can actually interact with the machines, etc. As it is, in testing, I can use these interactions while on the other side of the room, prone, etc.

SierraKomodo commented:
It's also worth noting the AI can use these as well - If that's intended then that's cool, just want to be sure you're aware.

KathyRyals commented:
Oops oh fuck oh god, tested PR :tm:

KathyRyals commented:
Fixed, but let the AI capabilities. Can change if asked to.

SierraKomodo reviewed (changes requested):
Summarizing the discussion in #coding

 - `Adjacent(user)` checks don't validate can use status, and will break remote functionality for AI and borgs
 - @CrimsonShrike recommends using `CanPhysicallyInteract(user)` or `CanInteract(user)` - The former will require adjacency, meaning borgs must be next to the scanners to eject and will probably block the AI. The latter just requires you to be able to interact with the machine, allowing borgs and AI to access it remotely.

SierraKomodo commented:
Also, you should squash your commits too once you're done.

KathyRyals commented:
Will change code and squash asap

KathyRyals commented:
Last commit : 

- Changed all the procs to check for CanPhysicallyInteract()
- Added alt click to Chem Dispenser and Chem Master, which ejects the beaker in your hand. This also makes it so that the UI button ejects the beaker in your hand. I have made this into a new proc in each machine called eject_beaker(mob/user).

SierraKomodo commented on code/game/machinery/Sleeper.dm:
```diff
@@ -256,6 +256,12 @@
 
+/obj/machinery/sleeper/AltClick(mob/user)
+	if(!CanPhysicallyInteract(user))
+		return
```
May want to consider `..()` instead of `return` so that other AltClick functionality can still be performed if they don't pass the adjacency check (I.e., AI's turf view).

SierraKomodo commented on code/game/machinery/bodyscanner.dm:
```diff
@@ -32,6 +32,11 @@
 
+/obj/machinery/bodyscanner/AltClick(mob/user)
+	if(!CanPhysicallyInteract(user))
+		return
```
Same as above

SierraKomodo commented on code/game/machinery/cryo.dm:
```diff
@@ -272,15 +272,28 @@
 
+/obj/machinery/atmospherics/unary/cryo_cell/AltClick(mob/user)
+	if(!CanPhysicallyInteract(user))
+		return
```
Same as above

SierraKomodo commented on code/modules/reagents/dispenser/dispenser2.dm:
```diff
@@ -166,13 +174,16 @@
+
+/obj/machinery/chemical_dispenser/AltClick(mob/user)
+	if(!CanPhysicallyInteract(user))
+		return
```
Also same

SierraKomodo commented on code/modules/reagents/Chemistry-Machinery.dm:
```diff
@@ -65,6 +65,21 @@
+
+/obj/machinery/chem_master/AltClick(mob/user)
+	if(!CanPhysicallyInteract(user))
+		return
```
Same (Wow there's a lot of these)

PsiOmegaDelta commented on code/game/machinery/cryo.dm:
```diff
@@ -272,15 +272,28 @@
 	return
 
+/obj/machinery/atmospherics/unary/cryo_cell/AltClick(mob/user)
+	if(!CanPhysicallyInteract(user))
```
Noting that this is inconsistent behavior with the UI/Topic, which does allow AI/borgs to eject people remotely

KathyRyals replied:
Will change !

PsiOmegaDelta commented on code/game/machinery/cryo.dm:
```diff
@@ -272,15 +272,28 @@
+	src.go_out()
+
+/obj/machinery/atmospherics/unary/cryo_cell/CtrlClick(mob/user)
+	if(Adjacent(user))
```
Any reason this is `Adjacent()` instead of `Can(Physically)Interact()`?

KathyRyals replied:
Forgot !

PsiOmegaDelta commented on code/modules/reagents/Chemistry-Machinery.dm:
```diff
@@ -65,6 +65,21 @@
+	icon_state = "mixer0"
+
+/obj/machinery/chem_master/AltClick(mob/user)
+	if(!CanPhysicallyInteract(user))
```
Same as with the cryo. Topic allows AIs/borgs to eject the beaker remotely, whereas this is limited to physical contact

KathyRyals replied:
Will change !

SierraKomodo commented on code/game/machinery/bodyscanner.dm:
```diff
@@ -32,6 +32,12 @@
 	add_fingerprint(usr)
 
+/obj/machinery/bodyscanner/AltClick(mob/user)
+	if(CanPhysicallyInteract(user))
```
Did you mean to leave this as `CanPhysicallyInteract()` or did you also intend to replace this with `CanDefaultInteract()`?

Spookerton replied:
boop - as above.

SierraKomodo commented on code/game/machinery/cryo.dm:
```diff
@@ -272,15 +272,29 @@
+		if(on == 1)
+			on = 0
+		else
+			on = 1
```
Just for future reference, you can shorthand lines 292-295 with `on = !on` - It'll flip values of `1`/`0` (`TRUE`/`FALSE`)

SierraKomodo reviewed (commented):
Also noticed that AI is able to use the CTRL+CLICK and ALT+CLICK shortcuts (Except for the body scanner which isn't using `CanDefaultInteract()` in this current build). Unsure if that was intended or not - If it was, that's fine, just wanted to make sure you were aware.

KathyRyals commented:
It's intended behaviour for both. Currently the AI can use the UI to do everything except ejecting people from scanner. I just wanted to introduce QoL changes so I didn't change it.

Spookerton reviewed (commented):
I know it's because of copypasting stuff around, but avoid `src.blah` in new code. Instance procs and vars are already in scope without it.

SierraKomodo reviewed (dismissed):
Squash

KathyRyals committed:
Added a bunch of alt clicks / ctrl click behaviours.


Fixed being able to eject from everywhere


Changed behaviour, added chem dispenser and chem master eject proc.


Fuck


Fixed again goddammit


Implemented suggested changes.


Travis man bad


WIP


Fixed typo


Should work now.


Added grinder alt click, ctrl click, ctrlalt click


FIXED BULLSHIT SPACE


Flip my shit


Removed src
